books/spiders/book_spider.py

```diff
@@ -0,0 +1,78 @@
+from typing import Iterable, Dict
+
+import scrapy
+from scrapy.http import Response
+
+BOOK_LIST_CSS = "li.col-xs-6"
+NEXT_PAGE_CSS = "li.next a::attr(href)"
+TITLE_CSS = "div.col-sm-6 h1::text"
+PRICE_CSS = "div.col-sm-6 p.price_color::text"
+AMOUNT_CSS = "div.col-sm-6 p.instock *::text"
+RATING_CSS = "div.col-sm-6 p.star-rating::attr(class)"
+CATEGORY_CSS = "ul.breadcrumb li"
+DESCRIPTION_CSS = "article.product_page p::text"
+UPC_CSS = "table.table tr"
+
+
+class BookSpiderSpider(scrapy.Spider):
+    name = "book_spider"
+    allowed_domains = ["books.toscrape.com"]
+    start_urls = ["https://books.toscrape.com/"]
+
+    def parse(self, response: Response, **kwargs) -> Response:
+        for book in response.css(BOOK_LIST_CSS):
+            detail_page = book.css(
+                "h3 a::attr(href)"
+            ).get()
+            yield response.follow(
+                detail_page, callback=self.parse_book
+            )
+
+        next_page = response.css(
+            NEXT_PAGE_CSS
+        ).get()
+        if next_page is not None:
+            yield response.follow(next_page, callback=self.parse)
```

Comment:
In the `parse` method, you are assuming that the `detail_page` and `next_page` will always be present in the response. However, it's a good practice to handle situations where these might not be present. You could add a check to see if `detail_page` and `next_page` are `None` and log an appropriate message if they are not present.